app/code/Magento/Backend/Api/Search/ItemsInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ *
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Backend\Api\Search;
+
+/**
+ * @api
+ */
+interface ItemsInterface
+{
+    const LIMIT = 'limit';
+    const START = 'start';
+    const QUERY = 'query';
+
+    /**
+     * get the search result items
+     *
+     * @return array
+     */
+    public function getResults();
+
+    /**
+     * set offset
+     *
+     * @param int $start
+     * @return ItemsInterface
```

Comment:
@maghamed I was trying the SearchCriteria approach, but hit a wall. 
I was doing it like this.  
have an `ItemInterface` with a single method `getResults(\Magento\Framework\Api\SearchCriteria $searchCriteria)` but I cannot add the filter in the search criteria for the query because the query filtering is dependent on the entity I'm filtering. So maybe I can do it via an additional parameter?. Have `getResults` look like this `getResults($searchCriteria, $query)` and the classes that implement this interface are responsible for handling the query.  
I could also avoid changing the `$searchCriteria` parameter by using an additional searchCriteriaBuilder inside the implementations of `getResults`. (this is already like that for the customer search)